### PR TITLE
[Strategy]Add StrategySpookySwapLPV2

### PR DESCRIPTION
**Pull request by merlincdj**

### CODE REVIEW

File: contracts/earn/strategies/SpookySwap/StrategyLPV2.sol [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        require(msg.sender == vault, 'StrategySpookySwapLPV2: NOT_VAULT');
+
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance < amount) {
+            IMasterChefV2(masterChefV2).withdraw(pid, amount.sub(wantBalance));
+            wantBalance = IERC20(want).balanceOf(address(this));
+        }
+
+        // just in case the decimal precision for the very left staking amount
+        uint256 withdrawAmount = amount < wantBalance ? amount : wantBalance;
```
Comment (fb-alexcq) [inline contracts/earn/strategies/SpookySwap/StrategyLPV2.sol:121]: require(wantBalance >= amount.mul(9999).div(10000), 'StrategySJoe: WITHDRAW_INSUFF_AMOUNT');
        uint256 withdrawAmount = amount < wantBalance ? amount : wantBalance;

I'd suggest to add this require . (Okay to go with 999/1000

Comment (fb-alexcq) [inline contracts/earn/strategies/SpookySwap/StrategyLPV2.sol:121]: require(wantBalance >= amount.mul(9999).div(10000), 'StrategySJoe: WITHDRAW_INSUFF_AMOUNT');
        uint256 withdrawAmount = amount < wantBalance ? amount : wantBalance;


Maybe better to add 'require' as above; could update to 999/1000.

File: contracts/earn/strategies/SpookySwap/StrategyLPV2.sol [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+            TransferHelper.safeTransfer(want, vault, wantBalance);
+        }
+    }
+
+    function emergencyExit() external override onlyAdmin {
+        IMasterChefV2(masterChefV2).emergencyWithdraw(pid, address(this));
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance > 0) {
+            TransferHelper.safeTransfer(want, vault, wantBalance);
+        }
```
Comment (fb-alexcq) [inline contracts/earn/strategies/SpookySwap/StrategyLPV2.sol:209]: What about ? (I'm not sure, just a suggestion)

          IMasterChefV2(masterChefV2).emergencyWithdraw(pid, vault);

Comment (fb-alexcq) [reply contracts/earn/strategies/SpookySwap/StrategyLPV2.sol:209]: Same for the method above.

**Review by fb-alexcq — COMMENTED** [review]
Two inline commments.

File: contracts/earn/strategies/SpookySwap/StrategyLPV2.sol (hunk 2 above)
Comment (merlincdj) [reply contracts/earn/strategies/SpookySwap/StrategyLPV2.sol:209]: It works, good suggestion!

**Review by fb-alexcq — APPROVED** [review]

**Pull request merged**